libs/pando/ui-sheet/src/document.ts

```diff
@@ -0,0 +1,24 @@
+import type { ReactNode } from 'react'
+import type { Conditional } from './conditional'
+import type { Field } from './field'
+import type { Header } from './header'
+
+export type Document = TextDocument | FieldsDocument | ConditionalDocument
+
+export interface TextDocument extends BaseDocument {
+  type: 'text'
+  text: ReactNode
+}
+export interface FieldsDocument extends BaseDocument {
+  type: 'fields'
+  fields: Field[]
+}
+export interface ConditionalDocument extends BaseDocument {
+  type: 'conditional'
+  conditional: Conditional
+}
+
+export interface BaseDocument {
```

Comment:
would it make sense for a document to be componsed of multiple of these elements?

for example, if i want a 
```
HEADER
text
fields
conditional
```
all under one document header(like we have for talents)

Reply:
![image](https://github.com/user-attachments/assets/8871a5f9-d326-41e7-be0b-70b6798ceb5e)
This structure is a 1:1 match/parity of current ui sheet
blue = TextDocument
red = FieldsDocument
yellow = ConditionalDocument

green = optional header

So im not sure what you meant by the HEADER over multiple text/fields/conditional. I believe that is just created within the UI itself, not necessarily in the sheet

Reply:
Oh i misunderstand, I thought a character sheet is one UISheet, but i realized its a map of UIsheets, like
`{ auto:UISheets[], ...}`. I guess the naming convention confused me. Then disregard my comments, I am fine with the format for now.

Reply:
A UI sheet is made up of an array or map of documents. I should adjust the shaping in sheet.ts prob